js/src/logger.ts

```diff
@@ -217,10 +217,11 @@ class BraintrustState {
   // (safely) dynamically cast it whenever retrieving the logger.
   public currentLogger: Logger<false> | undefined;
   public currentSpan: IsoAsyncLocalStorage<Span>;
-  // For operations not tied to a particular data object instance, we maintain a
-  // global BackgroundLogger which is refreshed anytime we log in with different
```

Comment:
I found that refreshing the entire logger every time we re-login can lead to multiple loggers floating around, because we'll create one logger before logging in at all, then another one after we log in. Since some records were ending up in the first logger and others in the second, we were no longer logging in deterministic order.